grida-server/src/main/java/fr/insalyon/creatis/grida/server/dao/H2DAOFactory.java

```
@@ -57,7 +58,7 @@ public class H2DAOFactory extends DAOFactory {
     private Connection connection;
 
     public static H2DAOFactory getInstance() {
-        if (instance == null) {
+        if (Configuration.getInstance().getFeatures().hasPool && instance == null) {
```

I get this is a sort of safeguard and that is shouldn't be called.
In this case I would throw an exception in this case.

---

The alternative would be to throw a runtimeException, like `IllegalStateException`.
It would be like a NullPointerException but with an explicit message and some logs.
As it is not meant to happen, it's fine to me.

---

`IllegalStateException` done in https://github.com/virtual-imaging-platform/GRIDA/pull/45/commits/2c17a787ba255c3716eaf5403a5d2845d6e64021